kokkos+cuda build fails: invalid compiler for CUDA

Building `kokkos@3.2.00` with `+cuda` fails using `spack@develop` (0924b209847e2d7284ff0ff2b234f8afd47e0921 from `Fri Dec 11 14:16:28 2020 +0000`) 

System info:
* `Ubuntu 18.04` for `x86_64`
* `GCC 7.5.0`

Concrete spec: [kokkos-stffzt.yml.txt](https://github.com/spack/spack/files/5682051/kokkos-stffzt.yml.txt)
Build log: [kokkos-cuda-build-out.txt](https://github.com/spack/spack/files/5682073/kokkos-cuda-build-out.txt)

```
$> spack install -f ./kokkos-stffzt.yml
...
==> Installing kokkos-3.2.00-f7yr3hkhgd4gwcmm3nkplpsfewjple7y
==> Fetching https://github.com/kokkos/kokkos/archive/3.2.00.tar.gz
############################################################################################################################################################## 100.0%     -=O=-       #       #         #           #
==> kokkos: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/kokkos-3.2.00-f7yr3hkhgd4gwcmm3nkplpsfewjple7y' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/kokkos-3.2.00-f7yr3hkhgd4gwcmm3nkplpsfewjple7y/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/kokkos-3.2.00-f7yr3hkhgd4gwcmm3nkplpsfewjple7y/lib64;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libxml2-2.9.10-yn2r3wfhiilelyulh5toteicdtxjhw7d/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libiconv-1.16-jearpk4xci4zc7dkrza4fufaqfkq7rfl/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/xz-5.2.5-komekkmyciga3kl24edjmredhj3uyt7v/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/zlib-1.2.11-smoyzzo2qhzpn6mg6rd3l2p7b23enshg/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cuda-10.2.89-tk4ck435miv7w2ek73636ticlylg2s62/lib64' '-DCMAKE_PREFIX_PATH:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cmake-3.18.4-q6lka2f3tzaizib7p4ot6ijzaustqcgv;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cuda-10.2.89-tk4ck435miv7w2ek73636ticlylg2s62' '-DKokkos_ARCH_VOLTA70=ON' '-DKokkos_ENABLE_CUDA=ON' '-DKokkos_ENABLE_OPENMP=OFF' '-DKokkos_ENABLE_PTHREAD=OFF' '-DKokkos_ENABLE_SERIAL=ON' '-DKokkos_ENABLE_HIP=OFF' '-DKokkos_ENABLE_AGGRESSIVE_VECTORIZATION=OFF' '-DKokkos_ENABLE_COMPILER_WARNINGS=OFF' '-DKokkos_ENABLE_CUDA_LAMBDA=OFF' '-DKokkos_ENABLE_CUDA_LDG_INTRINSIC=OFF' '-DKokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE=OFF' '-DKokkos_ENABLE_CUDA_UVM=OFF' '-DKokkos_ENABLE_DEBUG=OFF' '-DKokkos_ENABLE_DEBUG_BOUNDS_CHECK=OFF' '-DKokkos_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK=OFF' '-DKokkos_ENABLE_DEPRECATED_CODE=OFF' '-DKokkos_ENABLE_EXAMPLES=OFF' '-DKokkos_ENABLE_EXPLICIT_INSTANTIATION=OFF' '-DKokkos_ENABLE_HPX_ASYNC_DISPATCH=OFF' '-DKokkos_ENABLE_PROFILING=ON' '-DKokkos_ENABLE_PROFILING_LOAD_PRINT=OFF' '-DKokkos_ENABLE_QTHREAD=OFF' '-DKokkos_ENABLE_TESTS=OFF' '-DKokkos_ENABLE_HPX=OFF' '-DKokkos_ENABLE_HWLOC=OFF' '-DKokkos_ENABLE_NUMACTL=OFF' '-DKokkos_ENABLE_MEMKIND=OFF' '-DCMAKE_CXX_COMPILER=/opt/spack/lib/spack/env/gcc/g++' '-DKokkos_CXX_STANDARD=11' '-DBUILD_SHARED_LIBS=True' '/tmp/root/spack-stage/spack-stage-kokkos-3.2.00-f7yr3hkhgd4gwcmm3nkplpsfewjple7y/spack-src'

1 error found in build log:
     6     -- Check for working CXX compiler: /opt/spack/lib/spack/env/gcc/g++ - skipped
     7     -- Detecting CXX compile features
     8     -- Detecting CXX compile features - done
     9     -- Setting policy CMP0074 to use <Package>_ROOT variables
     10    -- The project name is: Kokkos
     11    -- Using -std=c++11 for C++11 standard as feature
  >> 12    CMake Error at cmake/kokkos_test_cxx_std.cmake:120 (MESSAGE):
     13      Invalid compiler for CUDA.  The compiler must be nvcc_wrapper or Clang, but
     14      compiler ID was GNU
     15    Call Stack (most recent call first):
     16      cmake/kokkos_tribits.cmake:201 (INCLUDE)
     17      CMakeLists.txt:166 (KOKKOS_SETUP_BUILD_ENVIRONMENT)
     18

See build log for details:
  /tmp/root/spack-stage/spack-stage-kokkos-3.2.00-f7yr3hkhgd4gwcmm3nkplpsfewjple7y/spack-build-out.txt
```

@jjwilke @junghans @balay @haampie

--- a/var/spack/repos/builtin/packages/kokkos/package.py
+++ b/var/spack/repos/builtin/packages/kokkos/package.py
@@ -217,6 +217,11 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
 
     def cmake_args(self):
         spec = self.spec
+
+        if spec.satisfies('~wrapper+cuda') and not spec.satisfies('%clang'):
+            raise InstallError("Kokkos requires +wrapper when using +cuda"
+                               "without clang")
+
         options = []
 
         isdiy = "+diy" in spec

